rtt/transports/corba/RemotePorts.cpp

```diff
@@ -194,6 +189,20 @@ RTT::base::ChannelElementBase::shared_ptr RemoteInputPort::buildRemoteChannelOut
     return corba_ceb;
 }
 
+bool RemoteInputPort::disconnect(PortInterface* port)
+{
+    //just check the type and not relay on
+    RemoteOutputPort *oPort = dynamic_cast<RemoteOutputPort* >(port);
+    if(oPort == NULL){
+        Logger::In in("RemoteInputPort::disconnect(PortInterface& port)");
+        log(Error) << "Port: " << this->getName() << " could not be disconnected from: " << oPort->getName()
+                << " because it is not of type: RemoteOutputPort" << endlog();
+        return false;
+    }
+
+    return dataflow->removeConnection(oPort->getName().c_str(), this->getDataFlowInterface(), this->getName().c_str());
```

Comment:
> @doudou, true you are right, could be simplified by just using "dataflow" here.

That is definitely NOT what I implied here.

My point is: if the dataflow interface on which you call removeConnection and the one you give to the method are the same, I believe that it implies that both ports are managed by the *same* dataflow interface. Meaning that they come from the same component. What am I missing ?